activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskAdminRuntimeImpl.java

```diff
@@ -132,4 +138,136 @@ public Task release(ReleaseTaskPayload releaseTaskPayload) {
         taskService.unclaim(releaseTaskPayload.getTaskId());
         return task(releaseTaskPayload.getTaskId());
     }
+    
+    @Override
+    public Task assign(AssignTaskPayload assignTaskPayload) {
+        // Validate that the task is visible by the currently authorized user
+        Task task;
+        try {
+            task = task(assignTaskPayload.getTaskId());
+        } catch (IllegalStateException ex) {
+            throw new IllegalStateException("The authenticated user cannot assign task" + assignTaskPayload.getTaskId() + " due it is not a candidate for it");
+        }
+        
+        // check that the task doesn't have an assignee
+        // If task is assigned, release it
+        if (task.getAssignee() != null && !task.getAssignee().isEmpty()) {
+            taskService.unclaim(assignTaskPayload.getTaskId());
+        }
+
+        taskService.claim(assignTaskPayload.getTaskId(),
+                          assignTaskPayload.getAssignee());
+        
+        return task(assignTaskPayload.getTaskId());
+    }
+    
+    @Override
+    public void addCandidateUsers(CandidateUsersPayload candidateUsersPayload) {      
+        Task task;
+        try {
+           task = task(candidateUsersPayload.getTaskId()); 
```

Comment:
@CTI777 same here.. the check is only on then User endpoints.. if you are admin you can execute the operation no matter what..